### PR TITLE
fix: rp history check password

### LINKED ISSUE #4079: [Crash] Error: Private key must be 32 bytes long
I was *doing something...*, then Mask reports an error.

> Private key must be 32 bytes long

Error stack:

<pre>Error: Private key must be 32 bytes long
    at Accounts.sign (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.web3-eth-accounts.js:793:15)
    at RedPacket (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.1950.js:1854:176)
    at Ch (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:171:137)
    at gk (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:294:29)
    at fk (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:282:22)
    at ek (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:281:369)
    at Vj (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:281:229)
    at Pj (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:270:476)
    at J (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/6674.js:19025:203)
    at MessagePort.R (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/6674.js:19026:123)</pre>

## Build info
- Version: 1.36.2
- NODE_ENV: production
- target: chromium
- build: stable
- architecture: web
- firefoxVariant: false
- resolution: desktop
- BUILD_DATE: 2021-08-16T11:46:43.666Z
- VERSION: v1.29.12-788-g27e365af6

## Git ():

27e365af6 (released) on tag "v1.36.2"
git@github.com:DimensionDev/Maskbook.git

### PR DESCRIPTION
**Pull request by zhouhanseng**
closes #4079

### CODE REVIEW

**Comment by zhouhanseng** [comment]
@DimensionDev/qa

**Review by UncleBill — APPROVED** [review]
The logic is a little bit complex. You might break it into multiple concepts, like

```ts
const isSameChain = parsedChainId === chainId
const isClaimable = !isExpired && !isEmpty && !isClaimed && isSameChain
const isPasswordValid = payload.password && payload.password !== 'PASSWORD INVALID'
const canClaim = isClaimable && isPasswordValid
```

That will be more friendly to review and understand.

File: packages/maskbook/src/plugins/ITO/SNSAdaptor/PoolInList.tsx [hunk 1]
```diff
@@ -136,6 +136,7 @@ export function PoolInList(props: PoolInListProps) {
 
     const canWithdraw = !isWithdrawn && (listOfStatus.includes(ITO_Status.expired) || noRemain)
 
+    // Note: After upgrade to Asymmetrical secret key in the future, `canSend` requires `password` exists.
     const canSend = !listOfStatus.includes(ITO_Status.expired) && !noRemain
     const progress = 100 * Number(new BigNumber(pool.total).minus(pool.total_remaining).dividedBy(pool.total))
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/SNSAdaptor/PoolInList.tsx:141]:
```suggestion
    const progress = 100 * Number.parseInt(new BigNumber(pool.total).minus(pool.total_remaining).dividedBy(pool.total).toString(), 10)
```

**Comment by AinuX** [comment]
Checked

**Review by UncleBill — APPROVED** [review]

File: packages/maskbook/src/locales/en-US.json [hunk 2]
```diff
@@ -392,6 +392,7 @@
     "plugin_red_packet_claiming": "Claiming...",
     "plugin_red_packet_refund": "Refund",
     "plugin_red_packet_empty": "Empty",
+    "plugin_red_packet_data_broken": "The red packet can’t be sent due to data damage. Please withdraw the assets after {{duration}}.",
```
Comment (guanbinrui) [inline packages/maskbook/src/locales/en-US.json:395]:
```suggestion
    "plugin_red_packet_data_broken": "The red packet cannot be sent due to data damage. Please withdraw the assets after {{duration}}.",
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**